pandas/_testing.py

```diff
@@ -1702,8 +1710,12 @@ def assert_frame_equal(
     else:
         for i, col in enumerate(left.columns):
             assert col in right
-            lcol = left.iloc[:, i]
-            rcol = right.iloc[:, i]
+            lcol = left._ixs(i, axis=1)
```

Comment:
why are you changing this? this is not a public function and shouldn't be used here at all

Reply:
@jbrockmendel suggested that it would be a marginally faster option.
Looks like it actually is.

Reply:
it doesn't matter if its faster as its not public; these are heavily used public function (yes this is an implementation), but we want to use iloc here (and i doubt this actually makes any real world difference).